worlds/cv64/client.py

```diff
@@ -0,0 +1,168 @@
+
+    async def validate_rom(self, ctx: BizHawkClientContext) -> bool:
+        from CommonClient import logger
+
+        try:
+            # Check if ROM is some version of Castlevania 64
+            game_name = ((await bizhawk.read(ctx.bizhawk_ctx, [(0x20, 0x14, "ROM")]))[0]).decode("ascii")
+            if game_name != "CASTLEVANIA         ":
+                return False
+            
+            # Check if we can read the slot name. Doing this here instead of set_auth as a protection against
+            # validating a ROM where there's no slot name to read.
+            try:
+                slot_name_bytes = (await bizhawk.read(ctx.bizhawk_ctx, [(0xBFBFE0, 0x10, "ROM")]))[0]
+                self.rom_slot_name = bytes([byte for byte in slot_name_bytes if byte != 0]).decode("utf-8")
+            except UnicodeDecodeError:
+                logger.info("Could not read slot name from ROM. Are you sure this ROM matches this client version?")
+                return False
+        except UnicodeDecodeError:
+            return False
+        except bizhawk.RequestFailedError:
+            return False  # Should verify on the next pass
+
+        ctx.game = self.game
+        ctx.items_handling = 0b001
+        ctx.want_slot_data = True
+        ctx.watcher_timeout = 0.125
+        return True
+
+    async def set_auth(self, ctx: BizHawkClientContext) -> None:
+        ctx.auth = self.rom_slot_name
+
+    def on_package(self, ctx: BizHawkClientContext, cmd: str, args: dict) -> None:
+        if cmd == "Bounced":
+            if "tags" in args:
+                if "DeathLink" in args["tags"] and args["data"]["source"] != ctx.slot_info[ctx.slot].name:
+                    self.received_deathlinks += 1
+
+    async def game_watcher(self, ctx: BizHawkClientContext) -> None:
+
+        try:
+            read_state = await bizhawk.read(ctx.bizhawk_ctx, [(0x342084, 4, "RDRAM"),
+                                                              (0x389BDE, 6, "RDRAM"),
+                                                              (0x389BE4, 224, "RDRAM"),
+                                                              (0x389EFB, 1, "RDRAM")])
+
+            game_state = int.from_bytes(read_state[0], "big")
+            save_struct = read_state[2]
+            written_deathlinks = int.from_bytes(bytearray(read_state[1][4:6]), "big")
+            deathlink_induced_death = int.from_bytes(bytearray(read_state[1][0:1]), "big")
+            cutscene_value = int.from_bytes(read_state[3], "big")
+            num_received_items = int.from_bytes(bytearray(save_struct[0xDA:0xDC]), "big")
+
+            # Make sure we are in the Gameplay or Credits states before detecting sent locations and/or DeathLinks.
+            # If we are in any other state, such as the Game Over state, set currently_dead to false, so we can once
+            # again send a DeathLink once we are back in the Gameplay state.
+            if game_state not in [0x00000002 or 0x0000000B]:
```

Comment:
I think you probably meant `[0x00000002, 0x0000000B]`.

`0x00000002 or 0x0000000B` evaluates to `2` because `2` is truthy.

Reply:
Hahaha, oops!